### PR TITLE
#167891586: User Role Settings

### PR DESCRIPTION
**Pull request by tvpeter**
#### What does this PR do?

- Gives super admin the privilege to assign roles to users

#### Description of Task to be completed?

- Modify users table to contain all roles

- Creates an endpoint for assigning users roles

- Ensure that only `super admin`s can access this endpoint

- Write test to cover user cases

#### How should this be manually tested?

- run `npm run devstart` to start the local development server after installing the dependencies and configuring the db

- Using postman, make a patch request to `/api/v1/user/:email/role` supplying authorization token and the role of the user.

#### Any background context you want to provide?

- The super admin should be able to assign and modify user roles.

#### What are the relevant pivotal tracker stories?

[#167891586](https://www.pivotaltracker.com/story/show/167891586)

#### Screenshots (if appropriate)

Screen Shot 2019-09-04 at 4 38 01 AM
Screen Shot 2019-09-04 at 4 38 09 AM

#### Questions:

-N/A

### CODE REVIEW

**Review by Pomile — COMMENTED** [review]
Good Job!

File: src/database/migrations/20190821105503-create-user.js [hunk 1]
```diff
@@ -73,4 +73,4 @@ module.exports = {
     }
   })),
   down: (queryInterface) => queryInterface.dropTable('Users')
-};
+};
```
Comment (thatkazuk1) [inline src/database/migrations/20190821105503-create-user.js]: Include a newline here Peter

File: src/controllers/Admin.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+   * @param {object} req - HTTP Request object
+   * @param {object} res - HTTP Response object
+   * @returns {object} object
+   */
+  static async updateUserRole(req, res) {
+    const { email } = req.params;
+    const { role } = req.body;
+
+    try {
+      const user = await User.findOne({ where: { email }, attributes: ['role', 'status'] });
```
Comment (sneakymaxy) [inline src/controllers/Admin.js]: Could you separate the ```user not found``` and ```duplicate update``` response

File: src/controllers/Admin.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+      if (!user || role === user.dataValues.role || user.dataValues.status !== 'active') {
+        return res.status(404).json(
```
Comment (max-wel) [inline src/controllers/Admin.js:23]: Could you separate the ```user not found``` and ```duplicate update``` response

**Review by max-wel — APPROVED** [review]

**Review by thatkazuk1 — APPROVED** [review]
Looks good to me

File: src/database/migrations/20190821105503-create-user.js (hunk 1 above)
Comment (tvpeter) [reply src/database/migrations/20190821105503-create-user.js]: okay.

**Review by femitj — APPROVED** [review]

**Review by daniellamarr — APPROVED** [review]
Looks good!

**Pull request merged**